### DISCUSSION TITLE
[Forms][QML Widget] What are the default handlers for buttons in QML widgets?

### DISCUSSION THREAD

**Discussion by jonnyforestGIS** [opengisch/QField#4785] [Q&A]
Hi, 
based on this example:
https://docs.qfield.org/how-to/attributes-form/

```qml
import QtQuick 2.0
import QtQuick.Controls 2.0

Button {
    width: 200
    height: width/5
    text: "Open in Maps"
    onClicked: {
        Qt.openUrlExternally(expression.evaluate("'geo:0,0?q=' || $y || ',' || $x"));
    }
}

```

I want to modify the handler to open an installed android app from the Portuguese Government. I try to use this pattern:

```qml
Qt.openUrlExternally("intent://#Intent;package=pt.ama.autenticacaogov;end");
```

Probably I'm doing something wrong or this kind of handler doesn't work for this type of government app. If it doesn't work, what kind of handlers can we use or are available in this kind of Qt.openUrlExternally() ?

At this time I test with success the following handlers:

```qml

Qt.openUrlExternally("https://www.example.com");

Qt.openUrlExternally("mailto:someone@example.com");

Qt.openUrlExternally("geo:0,0?q=latitude,longitude");

Qt.openUrlExternally("tel:+123456789");

```

Best regards,
João  Gaspar

**Comment by lp-dj** [comment]
I would also be very interested in that. I've already tried a lot of things. But I can't manage to open an app directly.

The only way I know is via the standard applications for certain URI schemes. Maybe you are lucky and can set the desired app as the default application for a specific URI scheme.

Otherwise I can only think of the very clumsy and not very elegant way via the Appstore, for example: `Qt.openUrlExternally("https://play.google.com/store/apps/details?id=pt.ama.autenticacaogov");`

Maybe it would be possible to implement the support of the [app-scheme](https://www.w3.org/TR/app-uri/) in QField? Would a registration in the [AndroidManifest.xml](https://github.com/opengisch/QField/blob/master/platform/android/AndroidManifest.xml.in) be sufficient for this?

Thinking one step further:

It would be nice if you could call a specific app with a particular action and have QField pick up the result of the process ;)

Similar to how it works with taking photos using an external photo app.

Scenario: Fill in a plant list for a biotope in QField, press the Qt button to open the app for automatic plant identification based on a photo, the result of the identification is transferred to QField, maybe even with the corresponding photo if required.
